platooning_control/CMakeLists.txt

```diff
@@ -12,112 +13,70 @@
-
-catkin_package(
-   INCLUDE_DIRS include
-   CATKIN_DEPENDS carma_utils cav_msgs roscpp std_msgs autoware_msgs
-)
+## Find dependencies using ament auto
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
 
-###########
-## Build ##
-###########
+# Name build targets
+set(node_exec platoon_control_node_exec)
+set(node_lib platoon_control_node)
 
-## Specify additional locations of header files
-## Your package locations should be listed before other locations
+# Includes
 include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
   ${Boost_INCLUDE_DIRS}
+  ${catkin_INCLUDE_DIRS}
+  include
 )
 
-file(GLOB_RECURSE headers */*.hpp */*.h)
-
-add_executable( ${PROJECT_NAME}
-  ${headers}
-  src/platoon_control.cpp
-  src/main.cpp)
-
-
-add_library(platoon_control_plugin_lib
- src/platoon_control.cpp
- src/platoon_control_worker.cpp
- src/pid_controller.cpp
- src/pure_pursuit.cpp
+# Build
+ament_auto_add_library(${node_lib} SHARED
+        src/platoon_control.cpp
+        src/platoon_control_worker.cpp
+        src/pid_controller.cpp
 )
 
-add_dependencies(platoon_control_plugin_lib ${catkin_EXPORTED_TARGETS})
-
-target_link_libraries(platoon_control_plugin_lib ${catkin_LIBRARIES} ${Boost_LIBRARIES})
-target_link_libraries(${PROJECT_NAME} platoon_control_plugin_lib)
-
-
-#############
-## Install ##
-#############
-
-
-install(DIRECTORY include
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-  FILES_MATCHING PATTERN "*.h"
-  PATTERN ".svn" EXCLUDE
+ament_auto_add_executable(${node_exec}
+        src/main.cpp
 )
 
-## Install C++
-install(TARGETS ${PROJECT_NAME} platoon_control_plugin_lib
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
+# Register component
+rclcpp_components_register_nodes(${node_lib} "platoon_control::PlatoonControlPlugin")
 
-# Mark other files for installation (e.g. launch and bag files, etc.)
-install(DIRECTORY
-  launch config
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+# All locally created targets will need to be manually linked
+# ament auto will handle linking of external dependencies
+target_link_libraries(${node_exec}
+        ${node_lib}
 )
 
+# Testing
+if(BUILD_TESTING)
 
-#############
-## Testing ##
-#############
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies() # This populates the ${${PROJECT_NAME}_FOUND_TEST_DEPENDS} variable
 
-catkin_add_gmock(${PROJECT_NAME}-test
+  ament_add_gtest(test_platoon_control
+  test/test_main.cpp
+  test/node_test.cpp
+  test/test_control.cpp
   test/test_pid.cpp
-  test/test_pure.cpp
   test/test_worker.cpp
-  test/test_control.cpp
-  test/test_main.cpp)
+  )
 
-if(TARGET ${PROJECT_NAME}-test)
-  target_link_libraries(${PROJECT_NAME}-test platoon_control_plugin_lib ${catkin_LIBRARIES})
-endif()
+  ament_target_dependencies(test_platoon_control ${${PROJECT_NAME}_FOUND_TEST_DEPENDS})
```

Comment:
apparently `ament_lint_auto_find_test_dependencies()` actually handles this part, so also not needed